### DISCUSSION TITLE
Reading files in Pocketbase as a framework with JS

### DISCUSSION THREAD

**Discussion by Fra146** [pocketbase/pocketbase#4165] [Q&A] [answered]
My validation rules are in a JSON file. How do I go about parsing it? The [documentation for $os](https://pocketbase.io/jsvm/functions/_os.readFile.html) is extremely lacking and all I get in return is a date and random numbers (which I think are characters).

```js
onRecordBeforeAuthWithPasswordRequest((e) => {

...

    let file = $os.readFile('properties.json')
    console.log(file) // This only returns a list with numbers separated by a comma, and there's also a timestamp at the start.

...

}, "officers")
```

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what type of validations your JSON file is consisting of, but if it is a valid JSON file then you can read it directly and get a JS plain object with [`let data = require("properties.json")`](https://pocketbase.io/docs/js-overview/#loading-modules) (_if it is not a JSON but a JS file then make sure that it is exported as CJS aka. has `module.exports = ...`_).

**Reply by Fra146** [reply]
I meant that my JSON file contains the validation rules to follow in the hook.

But yeah, this works perfectly, would suggest documenting it. I was fooled by the statement that the engine is extremely limited.

Thank you so much!
